openingd/dualopend.c

```diff
@@ -1467,6 +1467,52 @@ static bool run_tx_interactive(struct state *state,
 	return true;
 }
 
+/* If there's a failure, we reset the state to the last
+ * valid channel */
+static void revert_channel_state(struct state *state)
+{
+	struct tx_state *tx_state = state->tx_state;
+	struct amount_sat total;
+	struct amount_msat our_msats;
+	enum side opener = state->our_role == TX_INITIATOR ? LOCAL : REMOTE;
+
+	/* We've already checked this */
+	if (!amount_sat_add(&total, tx_state->opener_funding,
+			    tx_state->accepter_funding))
+		abort();
+
+	/* We've already checked this */
+	if (!amount_sat_to_msat(&our_msats,
+				state->our_role == TX_INITIATOR ?
+					tx_state->opener_funding :
+					tx_state->accepter_funding))
+		abort();
+
+	/* If this is the first time through, we can't do anything */
+	if (!tx_state)
+		return;
```

Comment:
We crash if this was true though?

Reply:
Right, in fact it's always set to something (we set the original in `main`).